sky/data/data_utils.py

```diff
@@ -194,6 +195,24 @@ def parallel_upload(source_path_list: List[str],
                 [bucket_name] * len(commands)))
 
 
+def get_gsutil_platform_flags() -> str:
+    """Returns platform-specific flags for gsutil.
+
+    Returns:
+        str; Platform-specific flags for gsutil. Includes a trailing space if it
+        is not empty.
+    """
+    if platform.system() == 'Darwin':
+        # Disable multiprocessing on Mac. Multithreading is still enabled.
+        # gsutil on Mac has a bug with multiprocessing that causes it to
+        # crash when uploading files. Related:
+        # https://bugs.python.org/issue33725
+        # https://github.com/GoogleCloudPlatform/gsutil/issues/464
+        return '-o "GSUtil:parallel_process_count=1" '
+    else:
+        return ''
```

Comment:
Should this be using the bash script to decide the platform, as the command might be executed remotely, while the `platform.system() == 'Darwin'` is checked locally?

Reply:
Ah good catch @Michaelvll  - I've changed the implementation. PTAL.

I spent some time on it and it was hard to add make a flag an in-line conditional (e.g., something like `gsutil -m $([[ "$(uname)" == "Darwin" ]] && echo "-o \"GSUtil:parallel_process_count=1\" " || echo "") rsync ...`), but this would require running `eval` before the command. However, that broke the `-x 'git/*'` flag for excluding, because now it would also evaluate `git/*` as a bash command instead of an arg.

Finally, I decided to keep it clean and use an if conditional to echo the right gsutil command, and run an eval on the echoed statement to run it as a command. 

Tested:
- [x] Code formatting: bash format.sh
- [x] Storage smoke tests: pytest tests/test_smoke.py::TestStorageWithCredentials
- [x] pytest tests/test_smoke.py::test_gcp_storage_mounts
- [x]  pytest tests/test_smoke.py::test_file_mounts